plugin/session_view.py

```diff
@@ -224,20 +226,25 @@ def _apply_auto_complete_triggers(
 
     def _increment_hover_count(self) -> None:
         settings = self.view.settings()
-        count = settings.get(self.HOVER_PROVIDER_COUNT_KEY, 0)
+        count = settings.get(HOVER_PROVIDER_COUNT_KEY, 0)
         if isinstance(count, int):
             count += 1
-            settings.set(self.HOVER_PROVIDER_COUNT_KEY, count)
-            settings.set(self.SHOW_DEFINITIONS_KEY, False)
+            settings.set(HOVER_PROVIDER_COUNT_KEY, count)
+            window = self.view.window()
+            if window and window.settings().get(HOVER_ENABLED_KEY, True):
+                settings.set(SHOW_DEFINITIONS_KEY, False)
 
     def _decrement_hover_count(self) -> None:
         settings = self.view.settings()
-        count = settings.get(self.HOVER_PROVIDER_COUNT_KEY)
+        count = settings.get(HOVER_PROVIDER_COUNT_KEY)
         if isinstance(count, int):
             count -= 1
             if count == 0:
-                settings.erase(self.HOVER_PROVIDER_COUNT_KEY)
-                settings.set(self.SHOW_DEFINITIONS_KEY, True)
+                settings.erase(HOVER_PROVIDER_COUNT_KEY)
+                self.reset_show_definitions()
+
+    def reset_show_definitions(self) -> None:
+        self.view.settings().set(SHOW_DEFINITIONS_KEY, globalprefs().get(SHOW_DEFINITIONS_KEY))
```

Comment:
It's not what we have done before either but wouldn't it make sense to just erase the `SHOW_DEFINITIONS_KEY` key? This should be more correct in case the global setting changes at alter point.

Reply:
Yeah, sounds right to me. Indeed if you manually set the view's settings, they seem to be persistant even when the global settings changes.